c7n/utils.py

```diff
@@ -343,3 +345,27 @@ def _f(*args, **kw):
             raise
     functools.update_wrapper(_f, f)
     return _f
+
+
+def int_from_value(raw_val, default=0):
+    if isinstance(raw_val, int):
+        return raw_val
+
+    try:
+        value = int(raw_val)
+    except ValueError:
+        value = default
+    return value
+
+
+def datetime_from_value(raw_val, default=None):
+    if isinstance(raw_val, datetime):
+        return raw_val
+
+    if default is None:
+        default = datetime.now(tz=tzutc())
```

Comment:
In what context would it break something? The existing calls to this utility are defined to replicate previous behavior, so there shouldn’t be any surprises there. And any future calls can be written to suit the need.

Is your concern that it uses “now” or that is uses something other than the integer 0? We could make the fallback `datetime.fromtimestamp(0)` instead if you think that’s less likely to cause a problem.

I don’t really like the idea of having the fallback be of a different type from the “success” value. That seems more likely to break things. (i.e. calling `isoformat()` on the result when the result is an integer)